src/firmware/app/control/trajectory_planner_impl.h

```diff
@@ -0,0 +1,275 @@
+ *
+ * @param max_allowable_acceleration [in] The maximum acceleration that can occur anywhere
+ * on the trajectory
+ *
+ * @param initial_speed [in] The initial speed at the beginning of the trajectory.  Units
+ * of parameters must be in consistent magnitudes, ie: distances in mm and acceleration in
+ * mm/s.
+ *
+ * @param num_segments [in] The number of nodes(elements) in the trajectory
+ *
+ * @param speeds [in/out] The existing speed profile to be modified in-place to be
+ * backwards continuous. This array is of length num_elements. The units of output are
+ * consistent with the input parameters.
+ *
+ * @return A status indicating whether or not modification was successful
+ */
+TrajectoryPlannerGenerationStatus_t
+app_trajectory_planner_modifySpeedsToBeBackwardsContinuous_impl(
+    unsigned int num_segments, float *segment_lengths, float max_allowable_acceleration,
+    float initial_speed, float *speeds);
+
+/**
+ * Function creates a forwards continuous speed profile based on the specified parameters.
+ * The profile will never exceed the maximums speeds specified by the
+ * max_allowable_speed_profile
+ *
+ * @pre All arrays must be pre-allocated up to at least
+ * TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS
+ *
+ * @param segment_lengths [in] The length of each segment between data points on the speed
+ * profile. In meters.
+ *
+ * @param max_allowable_speed_profile [in] An array that limits the max speed at each
+ * point on the profile. Units of parameters must be in consistent magnitudes, ie:
+ * distances in mm and speed in mm/s.
+ *
+ * @param max_allowable_acceleration [in] The maximum allowable acceleration along any
+ * segment in the profile. Units of parameters must be in consistent magnitudes, ie:
+ * distances in mm and acceleration in mm/s^2.
+ *
+ * @param initial_speed [in] The initial speed at the first element of the profile.  Units
+ * of parameters must be in consistent magnitudes, ie: distances in mm and speed in mm/s.
+ *
+ * @param final_speed  [in] The final speed at the last element of the profile.  Units of
+ * parameters must be in consistent magnitudes, ie: distances in mm and speed in mm/s.
+ *
+ * @param num_elements [in] The number of elements in the speed profile
+ *
+ * @param speeds [out] The pre-allocated array that will be modified to contain a forwards
+ * continuous speed profile that obeys the initial and final speeds and the
+ * max_allowable_speed_profile that limits the speed at each point on the trajectory.
+ * Units are consistent with the input parameters. This array if of length num_elements.
+ *
+ * @return A status indicating whether or not generation was successful
+ */
+TrajectoryPlannerGenerationStatus_t
+app_trajectory_planner_createForwardsContinuousSpeedProfile_impl(
+    unsigned int num_elements, float *segment_lengths, float *max_allowable_speed_profile,
+    float max_allowable_acceleration, float initial_speed, float final_speed,
+    float *speeds);
```

Comment:
similar comment here with `num_elements`

Reply:
Fixed 👍

Reply:
Don't see the change here either?

Reply:
Should be fixed now 👍

Reply:
Similar comment, `final_speed` seems to be out of position. Also it seems a bit odd that `initial_speed` and `final_speed` aren't beside each other, but that might just be me.